### PR TITLE
Feature/cloudinary image upload

### PR DESCRIPTION
**Pull request by dgaleano201611**
Summary
- adding the configuration of Cloudinary  for managing images in the application. 
- adding middleware formData in formData.ts file in midleware folder. 
- adding a POST request to the "/test-formdata" endpoint in app.ts

### CODE REVIEW

File: src/middleware/formData.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
+//importar
+import busboy from "busboy";
+import { v2 as cloudinary } from "cloudinary";
+import { Request, Response, NextFunction } from "express";
+
+
+cloudinary.config({
+  cloud_name: "dmshyev5s",
+  api_key: "295928881237629",
+  api_secret: "gItinS6-WkwZP45dZnfulJtnino",
```
Comment (ht1204) [inline src/middleware/formData.ts:10]: These values for cloudinary should be better in constant variables or in env variables, for safety.

File: src/middleware/formData.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
+//importar
```
Comment (ht1204) [inline src/middleware/formData.ts:1]: delete this commentary

File: src/middleware/formData.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  api_secret: "gItinS6-WkwZP45dZnfulJtnino",
+});
+
+export const formData = (req: Request, res: Response, next: NextFunction) => {
+  let uploadingFile = false;
+  let uploadingCount = 0;
+
+  const done = () => {
+    if (uploadingFile) return;
+    if (uploadingCount > 0) return;
```
Comment (ht1204) [inline src/middleware/formData.ts:19]: `if (uploadingCount) return;`

File: src/middleware/formData.ts [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+    next();
+  };
+
+  const bb = busboy({ headers: req.headers });
```
Comment (ht1204) [inline src/middleware/formData.ts:24]:
```js
const { headers } = req;
const bb = busboy({ headers });
```

**Comment by dgaleano201611** [comment]
- The environment variables were defined to configure the Cloudinary library in the application.
- To load the environment variables, the dotenv library had to be installed using npm and then the environment variables were loaded in this way:
 require("dotenv").config();

**Comment by dgaleano201611** [comment]
The name of the endpoint that uses the 'formdata' middleware in app.ts has been changed to: "/send-formdata"

**Review by ht1204 — APPROVED** [review]

**Pull request merged**